oauthenticator/azuread.py

```diff
@@ -38,6 +43,18 @@ def _tenant_id_default(self):
     def _username_claim_default(self):
         return 'name'
 
+    admin_azure_app_roles = List(
+        Unicode(),
+        config=True,
+        help="App roles that should give Jupyterhub admin privileges to a user",
+    )
+
+    allowed_azure_app_roles = List(
+        Unicode(),
+        config=True,
+        help="Automatically allow users with selected app roles",
+    )
+
```

Comment:
I think we should make these to be `Set` instead of `List`. Practically, if anyone passes a List, for example via a YAML configuration that is read and passed to the configuration - then it will be automatically casted to a Set so thats fine.

Reply:
I'd also like to have it clarified how to list the app roles. Is it their display name rather than id? That seems a bit odd to me but that is what matches your example configuration, but perhaps it is required to be unique as well or assumed to not risk causing problems.

Reply:
Perhaps a link or something as a reference to learn more about app roles this is relevant as well? I'm not sure, but in general I'm thinking the help text seem a bit too short and could be extended with another sentence or two.

Reply:
@consideRatio 
> I think we should make these to be Set instead of List. Practically, if anyone passes a List, for example via a YAML configuration that is read and passed to the configuration - then it will be automatically casted to a Set so thats fine.

In other implementations for IDPs, I saw both `Set` and `List` being used, so wasn't sure which of those was the best fit. I'll change that to `Set` as per your recommendation.

> I'd also like to have it clarified how to list the app roles. Is it their display name rather than id? That seems a bit odd to me but that is what matches your example configuration, but perhaps it is required to be unique as well or assumed to not risk causing problems.

It's rather `value` that gets sent in the token claim. It's allowed to have the same `displayName`, but not the `value`. For the latter, Azure AD will throw an error like this:
`Failed to update jupyterhub application. Error detail: It contains duplicate value. Please Provide unique value.`
IDs are used by Azure AD internally and they're not exposed in the respective token claim.

> Perhaps a link or something as a reference to learn more about app roles this is relevant as well? I'm not sure, but in general I'm thinking the help text seem a bit too short and could be extended with another sentence or two.

Sure, I'll think about what I can add here.

Reply:
Moved to `Set` and improved help messages in https://github.com/jupyterhub/oauthenticator/pull/446/commits/6dc41a6df1e479a7a7048b7ddc49630319da40ea